gst_all_1.gst-plugins-bad does not build on Darwin

**Describe the bug**
Building gst_all_1.gst-plugins-bad errors out with:
```
Project name: gst-plugins-bad
Project version: 1.16.2
Using 'CC_FOR_BUILD' from environment with value: 'clang'

meson.build:1:0: ERROR: Compiler clang can not compile programs.
```

**To Reproduce**
nix-shell -p gst_all_1.gst-plugins-bad

**Expected behavior**
should compile

**Notify maintainers**
@matthewbauer 

**Metadata**
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3`
 - channels(louis): `"home-manager, nixpkgs-21.03pre250093.0da76dab4c2"`
 - nixpkgs: `/Users/louis/.nix-defexpr/channels/nixpkgs`
```

Unrelated error when trying out the patch: 
```
./check_bindir "z$bindir" "z$execdir" "$bindir/git-add"
make: Entering directory '/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-git-2.29.2.drv-0/git-2.29.2/contrib/subtree'
make -C ../../ GIT-VERSION-FILE
make[1]: Entering directory '/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-git-2.29.2.drv-0/git-2.29.2'
make[1]: 'GIT-VERSION-FILE' is up to date.
make[1]: Leaving directory '/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-git-2.29.2.drv-0/git-2.29.2'
install -d -m 755 /nix/store/77d2m64lhbvyn97frdwcbnc0zhj7m0w2-git-2.29.2/libexec/git-core
install -m 755 git-subtree /nix/store/77d2m64lhbvyn97frdwcbnc0zhj7m0w2-git-2.29.2/libexec/git-core
asciidoctor -b docbook -d manpage  \
        -agit_version=2.29.2 -I../../Documentation -rasciidoctor-extensions -alitdd='&#x2d;&#x2d;' git-subtree.txt
`/homeless-shelter` is not a directory.
Bundler will use `/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-git-2.29.2.drv-0/bundler20201110-51332-nja1951332' as your home directory temporarily.
Traceback (most recent call last):
        9: from /nix/store/jyrmsp8qf6g97rkm9vpg63295l8s3phy-asciidoctor-2.0.10/bin/asciidoctor:18:in `<main>'
        8: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler.rb:149:in `setup'
        7: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:20:in `setup'
        6: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:101:in `block in definition_method'
        5: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/definition.rb:226:in `requested_specs'
        4: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/definition.rb:237:in `specs_for'
        3: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/definition.rb:170:in `specs'
        2: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `materialize'
        1: from /nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:80:in `map!'
/nix/store/8dv7qr9mmi9s1ggpkal7b4wp6x214q6f-bundler-2.1.4/lib/ruby/gems/2.6.0/gems/bundler-2.1.4/lib/bundler/spec_set.rb:86:in `block in materialize': Could not find nokogiri-1.10.3 in any of the sources (Bundler::GemNotFound)
make: *** [Makefile:86: git-subtree.xml] Error 1
make: Leaving directory '/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-git-2.29.2.drv-0/git-2.29.2/contrib/subtree'
builder for '/nix/store/zxddfcgflyj3ilcnyasd3q56mjbsbwc7-git-2.29.2.drv' failed with exit code 2
cannot build derivation '/nix/store/w36wpn0myxvbdk75h9vkw0nbalbn9zhl-cargo-1.47.0.drv': 1 dependencies couldn't be built
building '/nix/store/d23pgry1qldkasxsm42i77z4az331qnw-graphviz-2.42.2.drv'...
cannot build derivation '/nix/store/6dyh48fxbg5a8krvn1nks4dik3503xyh-librsvg-2.50.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lwpjdy48a8cp9z90vj36yz56l282bf1k-gst-plugins-bad-1.18.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/lwpjdy48a8cp9z90vj36yz56l282bf1k-gst-plugins-bad-1.18.0.drv' failed
```

--- a/pkgs/development/libraries/gstreamer/bad/default.nix
+++ b/pkgs/development/libraries/gstreamer/bad/default.nix
@@ -116,16 +116,18 @@ in stdenv.mkDerivation rec {
     meson
     ninja
     pkgconfig
+    orc # for orcc
     python3
     gettext
     gobject-introspection
   ] ++ optionals stdenv.isLinux [
-    wayland-protocols
+    wayland # for wayland-scanner
   ];
 
   buildInputs = [
     gst-plugins-base
     orc
+    gobject-introspection
     faad2
     libass
     libkate
@@ -179,6 +181,7 @@ in stdenv.mkDerivation rec {
   ] ++ optionals stdenv.isLinux [
     bluez
     wayland
+    wayland-protocols
   ] ++ optionals (!stdenv.isDarwin) [
     # wildmidi requires apple's OpenAL
     # TODO: package apple's OpenAL, fix wildmidi, include on Darwin
@@ -281,6 +284,9 @@ in stdenv.mkDerivation rec {
     # but its meson build system does not declare the dependency.
     "-Dapplemedia=disabled"
   ];
+
+  # Argument list too long
+  strictDeps = true;
 
   postPatch = ''
     patchShebangs \

